lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp

```diff
@@ -374,14 +381,39 @@ int amf_smf_send(
   char imsi[IMSI_BCD_DIGITS_MAX + 1];
   protocol_configuration_options_t* msg_pco;
 
+  if (!msg) {
+    return rc;
+  }
   if (amf_cause != AMF_CAUSE_SUCCESS) {
-    rc = amf_send_pdusession_reject(
-        &reject_req, msg->payload_container.smf_msg.header.pdu_session_id,
+    rc = amf_pdu_session_establishment_reject(
+        ue_id, msg->payload_container.smf_msg.header.pdu_session_id,
         msg->payload_container.smf_msg.header.procedure_transaction_id,
         amf_cause);
     return rc;
   }
 
+  M5GSmCause cause = M5GSmCause::INVALID_CAUSE;
+  /**
+   * Cause #27 – Missing or unknown DNN
+   * the external DN because the DNN was not included
+   * although required or if the DNN could not be resolved.
+   * */
+  if (msg->dnn.dnn.empty()) {
+    cause = M5GSmCause::MISSING_OR_UNKNOWN_DNN;
+  }
+
+  /*
+  Cause #28 – Unknown PDU session type
+  the requested PDU session type could not be recognized or is not allowed.
+  */
+  M5GPduSessionType session_type = static_cast<M5GPduSessionType>(
+      msg->payload_container.smf_msg.msg.pdu_session_estab_request
+          .pdu_session_type.type_val);
+  if (M5GPduSessionType::UNSTRUCTURED == session_type ||
+      M5GPduSessionType::ETHERNET == session_type) {
+    cause = M5GSmCause::UNKNOWN_PDU_SESSION_TYPE;
+  }
+
```

Comment:
This function is getting bigger. Can you please split it like                                        _amf_smf_validate_params and then rest of regular functionality

Reply:
Two new functions are created from this function. 1. For 5GSM Causes - amf_smf_validate_ie_params() and 2. For 5GMM Cause - amf_smf_validate_context().